SalesforceCommerceCloud/pwa-kit
# Fix fetching of translation file - @W-13566139@

Related to PR #1256 

This PR would make sure that we're fetching the correct translation files, which are whatever that is in the active project. Do not search within the base template.

Todos:
- [x] investigate a way to avoid having translation logic in the SDK (see [thread](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1261#discussion_r1226972603))
- [x] fix the CI failing tests
- [ ] address any performance issues - what are they?
- [x] what to do about the multiple instances of CopyPlugin? see [comment](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1261#discussion_r1226914763)

## How to test the PR:
Let's test both the retail-react-app and the generated project.

In retail-react-app package:
1. `cd template-retail-react-app && npm start`
2. View-source the homepage and look for the serialized translation in the html 
3. Select a different locale in the footer
4. View source the resulting homepage and verify the serialized translation has changed accordingly

Then test the generated project:
1. `GENERATOR_PRESET=retail-react-app-demo node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/generated-project`
2. `cd /tmp/generated-project && npm start`
3. View-source the homepage and look for the serialized translation in the html 
4. Select a different locale in the footer
5. View source the resulting homepage and verify the serialized translation has changed accordingly
